src/decode/rangecoder.rs

```diff
@@ -28,9 +28,51 @@ where
         Ok(dec)
     }
 
+    pub fn from_parts(stream: &'a mut R, range: u32, code: u32) -> Self {
+        Self {
+            stream,
+            range,
+            code,
+        }
+    }
+
+    pub fn into_parts(self) -> (u32, u32) {
+        (self.range, self.code)
+    }
+
+    pub fn set(&mut self, range: u32, code: u32) {
+        self.range = range;
+        self.code = code;
+    }
+
+    pub fn range(&self) -> u32 {
+        self.range
+    }
+
+    pub fn code(&self) -> u32 {
+        self.code
+    }
+
+    pub fn buf(&mut self) -> io::Result<&[u8]> {
+        self.stream.fill_buf()
+    }
+
+    pub fn remaining(&mut self) -> io::Result<usize> {
+        Ok(self.buf()?.len())
```

Comment:
Is that correct? As far as I remember, `fill_buf` can return fewer bytes than will be available in total, so this function doesn't return the number of bytes remaining, but only a lower bound, which isn't clear from the function's name.

Reply:
I removed this function since it was misleading and only used in one place. We are checking the number of currently available bytes to determine whether `try_process_next()` should be called.